bin/wasm-node/javascript/src/index.js

```diff
@@ -79,6 +104,34 @@ export async function start(config) {
       // `expected` was pushed by the `addChain` method.
       // Resolve the promise that `addChain` returned to the user.
       expected.resolve({
+        health: () => {
+          // are we already waiting for a health message reply?
+          if (chainHealthRequestCallbacks.chainId) {
+            const reject = chainHealthRequestCallbacks.chainId.reject;
+            delete chainHealthRequestCallbacks.chainId;
+            return reject(new Error('Called health before pending health request replied'));
+          }
```

Comment:
Hmm... When the first request does eventually get replied to, it will just be passed through to the consumer as if it was a regular RPC response

Reply:
In SmoldotProvider that will just result in a debug log because it won't know about the ID: https://github.com/paritytech/substrate-connect/blob/master/packages/connect/src/SmoldotProvider/SmoldotProvider.ts#L147